test/libcommonserver/io/testcheckdirectoryiterator.cpp

```diff
@@ -0,0 +1,237 @@
+/*
+ * Infomaniak kDrive - Desktop
+ * Copyright (C) 2023-2024 Infomaniak Network SA
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "testio.h"
+
+#include <filesystem>
+#include "libcommonserver/utility/testutility.h"
+
+using namespace CppUnit;
+
+namespace KDC {
+
+void TestIo::testCheckDirectoryIterator() {
+    testCheckDirectoryIteratorNonExistingPath();
+    testCheckDirectoryIteratorExistingPath();
+    testCheckDirectoryIteratotNextAfterEndOfDir();
+    testCheckDirectoryIteratorPermission();
+    testCheckDirectoryIteratorUnexpectedDelete();
+}
+
+void TestIo::testCheckDirectoryIteratorNonExistingPath() {
+    // Check that the directory iterator returns an error when the path does not exist
+    {
+        IoError error;
+        const DirectoryIterator it("C:\\nonexistingpath", false, error);
+
+        CPPUNIT_ASSERT_EQUAL(IoError::IoErrorNoSuchFileOrDirectory, error);
+    }
+
+    // Check that the directory iterator returns an error when the path syntax is invalid
+    {
+        IoError error;
+        const DirectoryIterator it("C:\\nonexistingpath\\*\\", false, error);
+
+        CPPUNIT_ASSERT_EQUAL(IoError::IoErrorNoSuchFileOrDirectory, error);
+    }
+}
+
+
+void TestIo::testCheckDirectoryIteratorExistingPath() {
+    // Check that the directory iterator is valid when the path is an empty directory and return EOF on first call
+    {
+        const TemporaryDirectory temporaryDirectory;
+        const SyncPath path = temporaryDirectory.path / "chekDirIt/empty_dir";
+        std::filesystem::create_directories(path);
+
+        IoError error;
+
+        DirectoryIterator it(path, false, error);
+        CPPUNIT_ASSERT_EQUAL(IoError::IoErrorSuccess, error);
+
+        DirectoryEntry entry;
+        CPPUNIT_ASSERT_EQUAL(false, it.next(entry, error));
```

Comment:
For checking the value of boolean, you may simply use `CPPUNIT_ASSERT`.

Reply:
Thanks you, I applied this in commit b712a22.